<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SensorMonitor</class>
 <widget class="QMainWindow" name="SensorMonitor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sensor Data Monitor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="hierarchyTab">
       <attribute name="title">
        <string>Sensor Hierarchy</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="refreshButton">
          <property name="text">
           <string>Refresh Sensor List</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTreeWidget" name="sensorTree">
          <property name="headerHidden">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="statusLabel">
          <property name="text">
           <string>Status: Ready</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="plotTab">
       <attribute name="title">
        <string>Plot Data</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QGroupBox" name="selectionGroup">
          <property name="title">
           <string>Sensor Selection</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <layout class="QVBoxLayout" name="topicLayout">
             <item>
              <widget class="QLabel" name="topicLabel">
               <property name="text">
                <string>Topic:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="topicCombo"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="sensorLayout">
             <item>
              <widget class="QLabel" name="sensorLabel">
               <property name="text">
                <string>Sensors:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QListWidget" name="sensorList">
               <property name="selectionMode">
                <enum>QAbstractItemView::MultiSelection</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="timeGroup">
          <property name="title">
           <string>Time Range</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="startLabel">
             <property name="text">
              <string>Start Time:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDateTimeEdit" name="startTime">
             <property name="displayFormat">
              <string>yyyy-MM-dd hh:mm:ss</string>
             </property>
             <property name="calendarPopup">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="endLabel">
             <property name="text">
              <string>End Time:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDateTimeEdit" name="endTime">
             <property name="displayFormat">
              <string>yyyy-MM-dd hh:mm:ss</string>
             </property>
             <property name="calendarPopup">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <layout class="QHBoxLayout" name="timeButtonsLayout">
             <item>
              <widget class="QPushButton" name="lastHourButton">
               <property name="text">
                <string>Last Hour</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="last24HoursButton">
               <property name="text">
                <string>Last 24 Hours</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="plotButton">
          <property name="text">
           <string>Plot Selected Sensors</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>